src/sort.js

```diff
@@ -1,6 +1,12 @@
 import ascending from "./ascending.js";
+import permute from "./permute.js";
 
-export default function sort(values, comparator = ascending) {
+export default function sort(values, f = ascending) {
   if (typeof values[Symbol.iterator] !== "function") throw new TypeError("values is not iterable");
-  return Array.from(values).sort(comparator);
+  values = Array.from(values);
```

Comment:
Not exactly related to this commit, but aren't we losing something by using plain arrays?

In many cases (such as the optimal transport) I'm sorting a `Float32Array` of indices against values that I'm reading somewhere else.

To take a simple example, I want to sort all the pixels of an image based on their red value. I'd have `indices = Float32Array.from({length: w * h}, (_,i) => i)` and would want to do `sorted = d3.sort(indices, i => imageData.data[4*i])`. I'd rather have `sorted` be a Float32Array (and maybe the intermediate `f`, but I don't think it's generically possible since the accessor might return a string).

Reply:
We discussed type preservation of the input array previously in https://github.com/d3/d3-array/pull/170#issuecomment-691697028, and I still think it’s appropriate that these array methods always return plain arrays for generality and simplicity. If you want something fancier, you can call *array*.sort directly.